kit/scrolls/android/src/main/java/tonlabs/uikit/scrolls/UIKitAccordionOverlayView.java

```diff
@@ -0,0 +1,356 @@
+            }
+
+            @Override
+            public void onAnimationRepeat(Animator animation) {
+                // no-op
+            }
+        };
+
+        mEventDispatcher = UIManagerHelper.getEventDispatcherForReactTag(reactContext, getId());
+
+        /**
+         * Even though we disable scroll during animation
+         * ScrollView might change it's `scrollY` position.
+         * Imagine you had a big section that allow you to scroll,
+         * then when it's collapsed the size of the content would become lesser
+         * than the visible are. That force ScrollView to reset scroll position to 0
+         * to adjust content position.
+         * Here we intercept this moment and adjust position of the overlay.
+         */
+        mScrollListener = new ReactScrollViewHelper.ScrollListener() {
+            @Override
+            public void onScroll(ViewGroup scrollView, ScrollEventType scrollEventType, float xVelocity, float yVelocity) {
+                int scrollYDiff = prevScrollY - scrollView.getScrollY();
+                if (scrollYDiff == 0) {
+                    return;
+                }
+                adjustContainerPositionWhenScrollHappened(scrollYDiff);
+                prevScrollY = scrollView.getScrollY();
+                Log.d(REACT_CLASS, String.format("scrollY change during animation: %d", scrollYDiff));
+            }
+
+            @Override
+            public void onLayout(ViewGroup scrollView) {
+                int scrollYDiff = prevScrollY - scrollView.getScrollY();
+                if (scrollYDiff == 0) {
+                    return;
+                }
+                adjustContainerPositionWhenScrollHappened(scrollYDiff);
+                prevScrollY = scrollView.getScrollY();
+                Log.d(REACT_CLASS, String.format("scrollY change during animation: %d", scrollYDiff));
+            }
+        };
+    }
+
+    // MARK:- commands
+
+    public void show(ReadableArray args) {
+        int startY = (int) PixelUtil.toPixelFromDIP(args.getInt(0));
+        int endY = (int) PixelUtil.toPixelFromDIP(args.getInt(1));
+
+        /**
+         * Prevent scrolling during animation
+         * to make the whole thing easier to work with
+         * and remove edge cases
+         */
+        disableScrollViewIfAny();
+        /**
+         * See a description of `mScrollListener` for a rationale
+         */
+        listenToScrollChangesIfAny();
+
+        BitmapDrawable screenshot = takeScreenshot(startY, endY);
+
+        /**
+         * Just in case the command was fired before 'hide' was called
+         */
+        removeView(mOverlayContainer);
+
+        mOverlayImage.setImageDrawable(screenshot);
+        mOverlayImage.setScrollY(0);
+        int top = getImageTop(startY);
+        mOverlayContainer.measure(screenshot.getIntrinsicWidth(), screenshot.getIntrinsicHeight());
+        mOverlayContainer.layout(0, 0, screenshot.getIntrinsicWidth(), screenshot.getIntrinsicHeight());
```

Comment:
I don't know if it's an optimization or not, but it looks logical to me to find the size of the bitmap only once:
```suggestion
        int width = screenshot.getIntrinsicWidth();
        int height = screenshot.getIntrinsicHeight();
        mOverlayContainer.measure(width, height);
        mOverlayContainer.layout(0, 0, width, height);
```

Reply:
I actually think that no optimisation will be applied, as it just getters, there is no logic behind them.
In theory this code even may produce local allocation on stack for primitives (`int`s), but I guess JVM is smart enough to not allocate anything and just pass them through :)

Reply:
OK, I thought that since it's a pure bitmap it might need to do some calculations to get the size